### ISSUE TITLE
Zero grad for each sequence in batch?

### ISSUE THREAD

**Issue by WenchaoDing** [vvanirudh/social-lstm-pytorch#4]
Hi, 

I see that in your implementation you are zeroing grads for each sequence instead of each batch. Then the grads do not accumulate for the whole batch actually. I think it is not a mini-batch gradient descent, and I am confused about why you batch the data in your case? 

Thanks.
